tests failling
```
======================================================================
ERROR: test_get_number_of_cases_in_domain (corehq.apps.hqcase.tests.test_dbaccessors.DBAccessorsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/hqcase/tests/test_dbaccessors.py", line 38, in test_get_number_of_cases_in_domain
    get_number_of_cases_in_domain(self.domain),
  File "/mnt/commcare-hq-ro/corehq/apps/hqcase/analytics.py", line 21, in get_number_of_cases_in_domain
    return CaseES().domain(domain).size(0).run().total
  File "/mnt/commcare-hq-ro/corehq/apps/es/es_query.py", line 210, in run
    raw = run_query(self.index, self.raw_query, debug_host=self.debug_host)
  File "/mnt/commcare-hq-ro/corehq/elastic.py", line 164, in run_query
    return es_instance.search(es_meta.index, es_meta.type, body=q)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/__init__.py", line 531, in search
    doc_type, '_search'), params=params, body=body)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
    self._raise_error(response.status, raw_data)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
NotFoundError: TransportError(404, u'IndexMissingException[[test_hqcases_2016-03-04] missing]')

======================================================================
ERROR: test_get_number_of_forms_in_domain (couchforms.tests.test_analytics.CouchformsAnalyticsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/couchforms/tests/test_analytics.py", line 60, in test_get_number_of_forms_in_domain
    get_number_of_forms_in_domain(self.domain), 2)
  File "corehq/ex-submodules/couchforms/analytics.py", line 36, in get_number_of_forms_in_domain
    return FormES().domain(domain).size(0).run().total
  File "/mnt/commcare-hq-ro/corehq/apps/es/es_query.py", line 210, in run
    raw = run_query(self.index, self.raw_query, debug_host=self.debug_host)
  File "/mnt/commcare-hq-ro/corehq/elastic.py", line 164, in run_query
    return es_instance.search(es_meta.index, es_meta.type, body=q)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/__init__.py", line 531, in search
    doc_type, '_search'), params=params, body=body)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
    self._raise_error(response.status, raw_data)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
NotFoundError: TransportError(404, u'IndexMissingException[[test_xforms_2016-07-07] missing]')

```

--- a/corehq/apps/hqcase/tests/test_dbaccessors.py
+++ b/corehq/apps/hqcase/tests/test_dbaccessors.py
@@ -1,13 +1,33 @@
+import uuid
+
 from django.test import TestCase
+from elasticsearch.exceptions import ConnectionError
+from mock import patch
+
 from casexml.apps.case.dbaccessors import get_open_case_ids_in_domain
 from casexml.apps.case.models import CommCareCase
 from casexml.apps.case.util import create_real_cases_from_dummy_cases
-from corehq.apps.hqcase.analytics import get_number_of_cases_in_domain_of_type, \
-    get_number_of_cases_in_domain
-from corehq.apps.hqcase.dbaccessors import get_case_ids_in_domain, get_case_types_for_domain, get_cases_in_domain, \
-    get_case_ids_in_domain_by_owner, \
-    get_all_case_owner_ids, get_case_properties
 from couchforms.models import XFormInstance
+from pillowtop.es_utils import initialize_index_and_mapping
+from testapps.test_pillowtop.utils import process_kafka_changes, process_couch_changes
+
+from corehq.apps.hqcase.analytics import (
+    get_number_of_cases_in_domain_of_type,
+    get_number_of_cases_in_domain,
+)
+from corehq.apps.hqcase.dbaccessors import (
+    get_all_case_owner_ids,
+    get_cases_in_domain,
+    get_case_ids_in_domain,
+    get_case_ids_in_domain_by_owner,
+    get_case_properties,
+    get_case_types_for_domain,
+)
+from corehq.elastic import get_es_new, EsMeta
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from corehq.pillows.mappings.case_mapping import CASE_INDEX_INFO
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import trap_extra_setup, create_and_save_a_case
 
 
 class DBAccessorsTest(TestCase):
@@ -32,12 +52,6 @@ class DBAccessorsTest(TestCase):
     def tearDownClass(cls):
         CommCareCase.get_db().bulk_delete(cls.cases)
         XFormInstance.get_db().bulk_delete(cls.forms)
-
-    def test_get_number_of_cases_in_domain(self):
-        self.assertEqual(
-            get_number_of_cases_in_domain(self.domain),
-            len([case for case in self.cases if case.domain == self.domain])
-        )
 
     def test_get_number_of_cases_in_domain__type(self):
         self.assertEqual(
@@ -158,3 +172,37 @@ class DBAccessorsTest(TestCase):
              for prop in (action.updated_known_properties.keys() +
                           action.updated_unknown_properties.keys())}
         )
+
+
+class ESAccessorsTest(TestCase):
+    domain = 'hqadmin-es-accessor'
+
+    def setUp(self):
+        super(ESAccessorsTest, self).setUp()
+        with trap_extra_setup(ConnectionError):
+            self.elasticsearch = get_es_new()
+            initialize_index_and_mapping(self.elasticsearch, CASE_INDEX_INFO)
+
+    def tearDown(self):
+        ensure_index_deleted(CASE_INDEX_INFO.index)
+        FormProcessorTestUtils.delete_all_cases_forms_ledgers(self.domain)
+        super(ESAccessorsTest, self).tearDown()
+
+    @patch('corehq.apps.hqcase.analytics.CaseES.index', CASE_INDEX_INFO.index)
+    @patch('corehq.apps.es.es_query.ES_META', {CASE_INDEX_INFO.index: EsMeta(CASE_INDEX_INFO.index, 'case')})
+    @patch('corehq.elastic.ES_META', {CASE_INDEX_INFO.index: EsMeta(CASE_INDEX_INFO.index, 'case')})
+    def test_get_number_of_cases_in_domain(self):
+        cases = [self._create_case_and_sync_to_es() for _ in range(4)]
+        self.assertEqual(
+            get_number_of_cases_in_domain(self.domain),
+            len(cases)
+        )
+
+    def _create_case_and_sync_to_es(self):
+        case_id = uuid.uuid4().hex
+        case_name = 'case-name-{}'.format(uuid.uuid4().hex)
+        with process_kafka_changes('CaseToElasticsearchPillow'):
+            with process_couch_changes('DefaultChangeFeedPillow'):
+                create_and_save_a_case(self.domain, case_id, case_name)
+        self.elasticsearch.indices.refresh(CASE_INDEX_INFO.index)
+        return case_id, case_name

--- a/corehq/apps/hqcase/tests/test_dbaccessors.py
+++ b/corehq/apps/hqcase/tests/test_dbaccessors.py
@@ -174,6 +174,11 @@ class DBAccessorsTest(TestCase):
         )
 
 
+TEST_ES_META = {
+    CASE_INDEX_INFO.index: EsMeta(CASE_INDEX_INFO.index, CASE_INDEX_INFO.type)
+}
+
+
 class ESAccessorsTest(TestCase):
     domain = 'hqadmin-es-accessor'
 
@@ -189,8 +194,8 @@ class ESAccessorsTest(TestCase):
         super(ESAccessorsTest, self).tearDown()
 
     @patch('corehq.apps.hqcase.analytics.CaseES.index', CASE_INDEX_INFO.index)
-    @patch('corehq.apps.es.es_query.ES_META', {CASE_INDEX_INFO.index: EsMeta(CASE_INDEX_INFO.index, 'case')})
-    @patch('corehq.elastic.ES_META', {CASE_INDEX_INFO.index: EsMeta(CASE_INDEX_INFO.index, 'case')})
+    @patch('corehq.apps.es.es_query.ES_META', TEST_ES_META)
+    @patch('corehq.elastic.ES_META', TEST_ES_META)
     def test_get_number_of_cases_in_domain(self):
         cases = [self._create_case_and_sync_to_es() for _ in range(4)]
         self.assertEqual(

--- a/corehq/ex-submodules/couchforms/tests/test_analytics.py
+++ b/corehq/ex-submodules/couchforms/tests/test_analytics.py
@@ -170,6 +170,12 @@ class ExportsFormsAnalyticsTest(TestCase, DocTestMixin):
                     'my://crazy.xmlns/app']
         }])
 
+
+TEST_ES_META = {
+    XFORM_INDEX_INFO.index: EsMeta(XFORM_INDEX_INFO.index, XFORM_INDEX_INFO.type)
+}
+
+
 class CouchformsESAnalyticsTest(TestCase):
     domain = 'hqadmin-es-accessor'
 
@@ -185,8 +191,8 @@ class CouchformsESAnalyticsTest(TestCase):
         super(CouchformsESAnalyticsTest, self).tearDown()
 
     @patch('couchforms.analytics.FormES.index', XFORM_INDEX_INFO.index)
-    @patch('corehq.apps.es.es_query.ES_META', {XFORM_INDEX_INFO.index: EsMeta(XFORM_INDEX_INFO.index, XFORM_INDEX_INFO.type)})
-    @patch('corehq.elastic.ES_META', {XFORM_INDEX_INFO.index: EsMeta(XFORM_INDEX_INFO.index, XFORM_INDEX_INFO.type)})
+    @patch('corehq.apps.es.es_query.ES_META', TEST_ES_META)
+    @patch('corehq.elastic.ES_META', TEST_ES_META)
     def test_get_number_of_cases_in_domain(self):
         forms = [self._create_form_and_sync_to_es() for _ in range(4)]
         self.assertEqual(

--- a/corehq/ex-submodules/couchforms/tests/test_analytics.py
+++ b/corehq/ex-submodules/couchforms/tests/test_analytics.py
@@ -1,22 +1,32 @@
 import datetime
 import uuid
+
 from django.test import TestCase
+from mock import patch
 from requests import ConnectionError
 
-from corehq.elastic import get_es_new, send_to_elasticsearch
-from corehq.pillows.mappings.xform_mapping import XFORM_INDEX_INFO
-from corehq.util.elastic import ensure_index_deleted
-from corehq.util.test_utils import DocTestMixin, trap_extra_setup
 from couchforms.analytics import (
+    app_has_been_submitted_to_in_last_30_days,
     domain_has_submission_in_last_30_days,
-    get_number_of_forms_in_domain,
-    get_first_form_submission_received, get_last_form_submission_received,
-    app_has_been_submitted_to_in_last_30_days, update_analytics_indexes,
     get_all_xmlns_app_id_pairs_submitted_to_in_domain,
-    get_form_analytics_metadata, get_exports_by_form,
+    get_exports_by_form,
+    get_first_form_submission_received,
+    get_form_analytics_metadata,
+    get_last_form_submission_received,
+    get_number_of_forms_in_domain,
+    update_analytics_indexes,
 )
 from couchforms.models import XFormInstance, XFormError
 from pillowtop.es_utils import initialize_index_and_mapping
+from testapps.test_pillowtop.utils import process_kafka_changes, process_couch_changes
+
+from corehq.elastic import get_es_new, EsMeta, send_to_elasticsearch
+from corehq.form_processor.interfaces.processor import FormProcessorInterface
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from corehq.form_processor.utils import TestFormMetadata
+from corehq.pillows.mappings.xform_mapping import XFORM_INDEX_INFO
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import DocTestMixin, get_form_ready_to_save, trap_extra_setup
 
 
 class CouchformsAnalyticsTest(TestCase, DocTestMixin):
@@ -54,10 +64,6 @@ class CouchformsAnalyticsTest(TestCase, DocTestMixin):
     def test_domain_has_submission_in_last_30_days(self):
         self.assertEqual(
             domain_has_submission_in_last_30_days(self.domain), True)
-
-    def test_get_number_of_forms_in_domain(self):
-        self.assertEqual(
-            get_number_of_forms_in_domain(self.domain), 2)
 
     def test_get_first_form_submission_received(self):
         self.assertEqual(
@@ -163,3 +169,37 @@ class ExportsFormsAnalyticsTest(TestCase, DocTestMixin):
             'key': ['exports_forms_analytics_domain', self.app_id_2,
                     'my://crazy.xmlns/app']
         }])
+
+class CouchformsESAnalyticsTest(TestCase):
+    domain = 'hqadmin-es-accessor'
+
+    def setUp(self):
+        super(CouchformsESAnalyticsTest, self).setUp()
+        with trap_extra_setup(ConnectionError):
+            self.elasticsearch = get_es_new()
+            initialize_index_and_mapping(self.elasticsearch, XFORM_INDEX_INFO)
+
+    def tearDown(self):
+        ensure_index_deleted(XFORM_INDEX_INFO.index)
+        FormProcessorTestUtils.delete_all_cases_forms_ledgers(self.domain)
+        super(CouchformsESAnalyticsTest, self).tearDown()
+
+    @patch('couchforms.analytics.FormES.index', XFORM_INDEX_INFO.index)
+    @patch('corehq.apps.es.es_query.ES_META', {XFORM_INDEX_INFO.index: EsMeta(XFORM_INDEX_INFO.index, XFORM_INDEX_INFO.type)})
+    @patch('corehq.elastic.ES_META', {XFORM_INDEX_INFO.index: EsMeta(XFORM_INDEX_INFO.index, XFORM_INDEX_INFO.type)})
+    def test_get_number_of_cases_in_domain(self):
+        forms = [self._create_form_and_sync_to_es() for _ in range(4)]
+        self.assertEqual(
+            get_number_of_forms_in_domain(self.domain),
+            len(forms)
+        )
+
+    def _create_form_and_sync_to_es(self):
+        with process_kafka_changes('XFormToElasticsearchPillow'):
+            with process_couch_changes('DefaultChangeFeedPillow'):
+                metadata = TestFormMetadata(domain=self.domain)
+                form = get_form_ready_to_save(metadata, is_db_test=True)
+                form_processor = FormProcessorInterface(domain=self.domain)
+                form_processor.save_processed_models([form])
+        self.elasticsearch.indices.refresh(XFORM_INDEX_INFO.index)
+        return form, metadata

